chesster.js

```diff
@@ -636,115 +522,63 @@ controller.hears([
     });
 });
 
-controller.hears([
-    players.appendPlayerRegex("pairing", true)
-], [
-    'direct_mention',
-    'direct_message'
-], function(bot, message) {
-    bot_exception_handler(bot, message, function() {
-        bot.startPrivateConversation(message, function (response, convo) {
-            var requestingPlayer = users.getByNameOrID(message.user);
-            var player = players.getSlackUser(users, message);
-
-            preparePairingCompetitorMessage(requestingPlayer, player, function(response) {
-                convo.say(response);
-            });
+slack.hears(controller, {
+    patterns: [
+        players.appendPlayerRegex("pairing", true)
+    ],
+    message_types: [
+        'direct_mention', 'direct_message'
+    ],
+    config: config,
+}, function(bot, message) {
+    var targetPlayer = players.getSlackUser(users, message);
+    var deferred = Q.defer();
+    var allLeagues = league.getAllLeagues(config);
+    bot.startPrivateConversation(message, function (response, convo) {
+        Q.all(
```

Comment:
Q.all returns a promise which will wait until all of the given promises have completed before it completes. Similar to async.parallel except for promises.

The promise is being used for flow control here, yes. but that's not the only reason it's being used. promises are now used throughout this paradigm. Happy to chat more about my use of them if desired.